src/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj

```diff
@@ -84,44 +85,45 @@
-
-                        (map? metrics)
-                        (ringutils/json-response
-                         200 (ks/mapvals metrics-utils/get-mbean metrics))
-
-                        :else
-                        (ringutils/json-response
-                         400 "metrics request must be a JSON array, string, or object")))
-
-                    (catch JsonParseException e
-                      (ringutils/json-response 400 {:error (str e)})))))
-
-              (comidi/GET ["/" [#".*" :names]] []
-                (fn [{:keys [route-params] :as req}]
-                  (let [name (java.net.URLDecoder/decode (:names route-params))]
-                    (if-let [mbean (metrics-utils/get-mbean name)]
-                      (ringutils/json-response 200 mbean)
-                      (ringutils/json-response 404
-                                                (format "No mbean '%s' found" name)))))))))
-    #(ring-defaults/wrap-defaults % ring-defaults/api-defaults))))
+    (comidi/wrap-routes
+     (comidi/context path
+         (comidi/context "/v1"
+             (comidi/context "/mbeans"
+                 (comidi/GET "" []
+                   (fn [req]
+                     (ringutils/json-response 200
+                                              (metrics-utils/mbean-names))))
+               (comidi/POST "" []
+                 (fn [req]
+                   (try
+                     (let [metrics (with-open [reader (-> req :body io/reader)]
+                                     (json/parse-stream reader true))]
+                       (cond
+                         (seq? metrics)
+                         (ringutils/json-response
+                          200 (map metrics-utils/get-mbean metrics))
+
+                         (string? metrics)
+                         (ringutils/json-response
+                          200 (metrics-utils/get-mbean metrics))
+
+                         (map? metrics)
+                         (ringutils/json-response
+                          200 (ks/mapvals metrics-utils/get-mbean metrics))
+
+                         :else
+                         (ringutils/json-response
+                          400 (tru "metrics request must be a JSON array, string, or object"))))
+
+                     (catch JsonParseException e
+                       (ringutils/json-response 400 {:error (str e)})))))
+
+               (comidi/GET ["/" [#".*" :names]] []
+                 (fn [{:keys [route-params] :as req}]
+                   (let [name (java.net.URLDecoder/decode (:names route-params))]
+                     (if-let [mbean (metrics-utils/get-mbean name)]
+                       (ringutils/json-response 200 mbean)
+                       (ringutils/json-response 404
+                                                (tru "No mbean ''{0}'' found" name)))))))))
+     #(ring-defaults/wrap-defaults % ring-defaults/api-defaults))
+   i18n/locale-negotiator)))
```

Comment:
so it could be something like `(comp i18n/locale-negotatior #(ring-defaults/blah ...))`?